restore/data_test.go

```diff
@@ -34,7 +34,7 @@ var _ = Describe("restore/data tests", func() {
 			execStr := regexp.QuoteMeta("COPY public.foo(i,j) FROM PROGRAM 'cat <SEG_DATA_DIR>/backups/20170101/20170101010101/gpbackup_<SEGID>_20170101010101_3456.gz | gzip -d -c' WITH CSV DELIMITER ',' ON SEGMENT")
 			mock.ExpectExec(execStr).WillReturnResult(sqlmock.NewResult(10, 0))
 			filename := "<SEG_DATA_DIR>/backups/20170101/20170101010101/gpbackup_<SEGID>_20170101010101_3456.gz"
-			_, err := restore.CopyTableIn(context.Background(), connectionPool, "public.foo", "(i,j)", filename, false, 0)
+			_, err := restore.CopyTableIn(context.Background(), connectionPool, "public.foo", "(i,j)", filename, false, 0, false)
```

Comment:
Please consider adding two unit-tests with mocked GetResizeClusterInfo() for larger to smaller resize.
One for isReplicated=true with a check that query executed only once.
Another for isReplicated=false with a check that query still executed multiple times and we don't break something.

Anticipating the question of the obviousness of tests - we do the tests for us in the future..To keep the contract of the function and to avoid not obvious artifacts when syncing with upstream.